claims.go

```diff
@@ -242,6 +242,47 @@ func (c Claims) GetInstanceID() (*[]byte, error) {
 	return instID, nil
 }
 
+func (c *Claims) nonceByProfile(profile string) ([]byte, error) {
+	var nonce []byte
+	switch profile {
+	case PSA_PROFILE_1:
+		if c.LegacyNonce == nil {
+			return nil, errors.New("missing mandatory nonce")
+		}
+		nonce = *c.LegacyNonce
+	case PSA_PROFILE_2:
+		if c.Nonce == nil {
+			return nil, errors.New("missing mandatory nonce")
+		}
+		if c.Nonce.Len() != 1 {
+			return nil, errors.New("there must be exactly one nonce")
+		}
+		nonce = c.Nonce.GetI(0)
+	default:
+		return nil, fmt.Errorf("unknown profile: %s", profile)
+	}
+
+	if err := isPSAHashType(nonce); err != nil {
+		return nil, fmt.Errorf("invalid nonce: %w", err)
+	}
+
+	return nonce, nil
+}
+
```

Comment:
Would it not be a better to have a mirror API like SetNonce().

If the profile is NOT set already that returns an Error.

if the profile is set to right value and is legacy, we set it on legacy nonce.
If profile is set to right value and it is new profile, we set the correct variable profile?

This way user will have one seamless API, rather than worrying about which Nonce parameter to set internally and tend to make mistake in usage at first place?

Reply:
I agree it'd be a good addition, but it's unrelated to this change.

Could you please raise a feature request issue for this?

Reply:
So long as the suggestion is meaningful, I am happy to move on from here! I will make a feature request and introduce it here and approve the PR!

Reply:
Raised Issue #21 to track the change!

Reply:
thanks!